tests/test_resolver.py

```diff
@@ -284,6 +284,25 @@ def test_iter_dependencies_ignores_constraints(resolver, from_line):
         next(res._iter_dependencies(ireq))
 
 
+def test_iter_dependencies_after_combine_install_requirements(
```

Comment:
Could you add an integration test to `test_cli_compile.py` instead of unit test, in case we refactor out resolver internals later (maybe)?

Reply:
Probably as expected, it'll take me a bit more time to formulate that, but I can do that. I'd say let's keep the unit test too, since (at least with the current resolver) it should be easier (clearer/faster) to debug upon failure?

Reply:
@atugushev Done!